client/coral-plugin-permalinks/PermalinkButton.js

```diff
@@ -0,0 +1,90 @@
+import React, {PropTypes} from 'react';
+import onClickOutside from 'react-onclickoutside';
+const name = 'coral-plugin-permalinks';
+
+class PermalinkButton extends React.Component {
+
+  static propTypes = {
+    asset_id: PropTypes.string.isRequired,
+    comment_id: PropTypes.string.isRequired
+  }
+
+  constructor (props) {
+    super(props);
+    this.state = {popoverOpen: false, copySuccessful: null, copyFailure: null};
+    this.toggle = this.toggle.bind(this);
+    this.copyPermalink = this.copyPermalink.bind(this);
+  }
+
+  toggle () {
+    this.setState({popoverOpen: !this.state.popoverOpen});
+  }
+
+  handleClickOutside () {
+    this.setState({popoverOpen: false});
+  }
+
+  copyPermalink () {
+    this.permalinkInput.select();
+    try {
+      document.execCommand('copy');
+      this.setState({copySuccessful: true});
+    } catch (err) {
+      this.setState({copyFailure: true});
+    }
+
+    setTimeout(() => {
+      this.setState({copyFailure: null, copySuccessful: null});
+    }, 4500);
+  }
+
+  render () {
+    const publisherUrl = 'http://nytimes.com/';
```

Comment:
When embedding these comments, we can't rely on any concept of a base url. These comments may be on properties across sub domains or even domains.

Our best bet is to look at the url that the comment stream is on, use that and append something that says where the view goes to.